### PR TITLE
USJ Generation

### PR DESCRIPTION
**Pull request by kavitharaju**
This PR inlcudes
- **USJ Generation**: Implement `usfm_parser.to_usj()`. AST to USJ conversion. Conversion follows same logic as USX generation does, but not dependent on it.
- **A simple JSON schema for USJ**: Defining the common structure of USJ objects, with valid properties and their type. **Not** defining what-objects-are-valid-where like in an extensive grammar. 
- **Tests**: Older `to_dict()` had tests for checking successful conversion and check for presence of all markers in output. Both sets passing for new `to_usj()` as well. Also checking the generated USJ against the above JSON schema.
- **To List**: Re-wrote the older `to_list()` method which was dependant on the previous JSON output to work with new USJ format.
- **Code Refactor**: Older single file `usfm_parser.py` which contained USFMParser class and all methods for USX, dict and list generation replaced with 4 files(usx_generator.py, usj_generator.py, list_generator.py), using classes for each of the above, reducing the usfm_parser.py file from 894 to 179 lines of code.
- **Documentation**: Updated details in architecture.md with details in code refactor. Updated Readme.md and the jupyter notebook in docs relpacing old `to_dict()` with new `to_usj()` 

:warning: This doesn't include implementation of "filter" feature, which was implemented partially in older `to_list()`

### CODE REVIEW

File: python-usfm-parser/README.md
Comment (joelthe1) [inline python-usfm-parser/README.md:1]: How about we call the parent directory as `py-usfm-parser`?

File: python-usfm-parser/src/usfm_grammar/usj_generator.py
Comment (joelthe1) [inline python-usfm-parser/src/usfm_grammar/usj_generator.py:1]: I suspect that we could improve some readability using the `match` case. But we can take a look into that in a future version.

Comment (kavitharaju) [reply python-usfm-parser/src/usfm_grammar/usj_generator.py:1]: image
Its not only `==` that we use, we have mixed comparisons with `in` , `endswith()` etc which I think are not supported by match-case. Hence fallen back to the old-fashioned if-else ladder. But we can revisit this later and check if match case can somehow handle such comparisons too.

**Comment by kavitharaju** [comment]
New changes
1. Folder renamed to py-usfm-grammar. And made subsequent changes in code and docs.
2. Version Updates
       - Moved to latest version of node and npm
       - Latest version of tree-sitter-cli npm package
       - Latest version of tree-sitter python package

**Pull request merged**